### PR TITLE
CLOUDP-164203: Improve private endpoint tests

### PR DESCRIPTION
**Pull request by helderjs**
### All Submissions:

* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).
* [ ] Update docs/release-notes/release-notes.md if your changes should be included in the release notes for the next release.

### CODE REVIEW

**Review by fabritsius — APPROVED** [review]
Looks Good 👍

File: test/e2e/api/aws/aws.go [hunk 1]
```diff
@@ -98,24 +98,28 @@ func (s sessionAWS) DeleteVPC(vpcID string) error {
...
-	if len(result.Subnets) < 1 {
-		return "", errors.New("can not find Subnet")
+
+	if len(result.Subnets) > 0 {
+		return *result.Subnets[0].SubnetId, nil
 	}
-	fmt.Println(result)
-	return *result.Subnets[0].SubnetId, nil
+
+	return s.CreateSubnet(vpcID, "10.0.0.0/24", "pe-aws-test")
```
Comment (igor-karpukhin) [inline test/e2e/api/aws/aws.go:122]: Is this created subnet gets automatically removed after the test?

Comment (igor-karpukhin) [reply test/e2e/api/aws/aws.go:122]: upd: it will be removed automatically by the cleaner

**Review by igor-karpukhin — APPROVED** [review]
Looks good 👍

**Pull request merged**